### PR TITLE
Cart

### PR DESCRIPTION
**Pull request by jhgould**
Add in user story 23 and 24. Definitely some things I want to go over with you both. I know there are better ways to do some of the actions so its a good opportunity for refactoring.

### CODE REVIEW

File: app/controllers/cart_controller.rb [hunk 1]
```diff
@@ -6,6 +6,21 @@ def add_item
...
+      cart.add_quantity(quantity, item, inventory)
+      redirect_to '/cart'
+  end
+
+  def decrease_quantity
+    item = Item.find(params[:item_id])
+    quantity = params[:quantity_decrease].to_i
+    cart.remove_quantity(quantity, item)
+    redirect_to '/cart'
+  end
```
Comment (jhgould) [inline app/controllers/cart_controller.rb:22]: These are very repetitive and I can definitely clean it up. I could use if statements to determine if the user is adding or subtracting from the quantity. A good refactor opportunity. But for now, it works.

File: app/models/cart.rb [hunk 2]
```diff
@@ -10,6 +10,11 @@ def add_item(item)
     @contents[item] += 1
   end
 
+  def remove_item(item)
+    @contents[item] = 0 if !@contents[item]
+    @contents[item] -= 1
+  end
+
```
Comment (jhgould) [inline app/models/cart.rb:17]: Literly the same as `add_item` but instead of the + I use a minus -.

File: app/models/cart.rb [hunk 3]
```diff
@@ -32,4 +37,18 @@ def total
...
+      quantity.times do
+        add_item(item.id.to_s)
+      end
+    end
+  end
+
+  def remove_quantity(quantity, item)
+    quantity.times do
+      remove_item(item.id.to_s)
+    end
```
Comment (jhgould) [inline app/models/cart.rb:51]: I created a loop to either add or subtract quantity based on user input. quantity is passed in through the controller and that determines out many times the loop runs.

File: app/views/cart/show.html.erb [hunk 4]
```diff
@@ -31,6 +31,14 @@
...
         <center>
           <p><%=quantity%></p>
+          <%= form_tag "/cart/#{item.id}/add_quantity", method: :patch do %>
+          <p><%= number_field_tag 'quantity', '1', min: 1, max: "#{item.inventory - 1}" %></p>
+          <%= submit_tag 'Update Quantity' %>
+          <% end %>
+          <%= form_tag "/cart/#{item.id}/decrease-quantity", method: :patch do %>
+          <p><%= number_field_tag 'quantity_decrease', '1', min: 1, max: "#{item.inventory - 1}" %></p>
+          <%= submit_tag 'Decrease Quantity' %>
+          <% end %>
```
Comment (jhgould) [inline app/views/cart/show.html.erb:41]: I think I could find a way to create a drop-down menu that would all the user to add or subtract quantity in the same form. For now, I have two separate forms.

File: app/controllers/cart_controller.rb [hunk 5]
```diff
@@ -6,6 +6,21 @@ def add_item
     redirect_to "/items"
   end
 
+  def add_quantity
```
Comment (EricLarson2020) [inline app/controllers/cart_controller.rb:9]: looks good

**Pull request merged**